test/MerkleProof.js

```diff
@@ -0,0 +1,60 @@
+var MerkleProof = artifacts.require("./MerkleProof.sol");
+
+import { sha3 } from "ethereumjs-util";
+import MerkleTree from "./helpers/merkleTree.js";
+
+contract('MerkleProof', function(accounts) {
+  let merkleProof;
+
+  before(async function() {
+    merkleProof = await MerkleProof.new();
+  });
+
+  describe("verifyProof", function() {
+    it("should return true for a valid Merkle proof", async function() {
+      const elements = ["a", "b", "c", "d"].map(el => sha3(el));
```

Comment:
Good point, I think that makes sense from a usability standpoint. A user can generate a merkle tree off-chain by passing in an array of elements and then when the user wants to submit a proof the user uses the same merkle tree to generate a proof and submits it along with the hash of the element to be proved